### DISCUSSION TITLE
Filtering containers by environment variable or name regex

### DISCUSSION THREAD

**Discussion by mateuszdrab** [containrrr/watchtower#1209] [Q&A] [answered]
I'm implementing K8s on top of docker and I want to prevent Watchtower from touching those containers whilst still updating all preexisting docker containers.

I could just use label enable feature but I am wondering if it is possible to instead exclude K8s containers. Unfortunately, it seems K8s pod labels don't propagate to docker level.

Is it possible to select/ignore containers to update by regex of their name or by environment variable instead of labels?

Thanks

**Comment by piksel** [comment, marked as answer]
No, only labels are supported for explicit exclusion. The only alternative is to explicitly *include* your other containers (either through command line or labels), or finding a way to add labels to your containers.

**Reply by mateuszdrab** [reply]
> No, only labels are supported for explicit exclusion. The only alternative is to explicitly _include_ your other containers (either through command line or labels), or finding a way to add labels to your containers.

Thanks for the answer.

It would have been too much work trying to find out why kubelet doesn't add the labels to the pods' containers
I have forked and modified Watchtower to enable regex in container name, works a treat.
Should I PR this?

Literally a few lines of code.
https://github.com/mateuszdrab/watchtower/commit/5999d0b6d2359d1dcca8e24125501b9c1a4a9191

**Reply by mateuszdrab** [reply]
Any update?

I think the regex feature could benefit some users

**Reply by simonaronsson** [reply]
great idea! I would suggest hiding it behind a feature flag, but other than that, I'm all in favor of adding this capability!

**Comment by mateuszdrab** [comment]
This has been merged https://github.com/containrrr/watchtower/pull/1241
